### ISSUE TITLE
ZFS immutable properties fails on re-run

### ISSUE THREAD

**Issue by techmunk** [ansible/ansible-modules-extras#769]
##### Ansible Configuration:

Pipelining enabled
##### Summary:

It's my understanding that Ansible is intended to be immutable. However, if you pass options to the ZFS module on one, those same options can actually cause a failure on a subsequent run, and the whole playbook to abort.
These properties are 'casesensitivity', 'normalization', and 'utf8only' (volblocksize should be added to this list, though that's likely another issue).
##### Steps To Reproduce:

The following task will work once when you don't yet have the ZVOL that is being created, but will fail on the next run with the message "Cannot change property %s after creation.".

``` yml
- name: ZFS Mountpoints
  zfs: state=present name=ztank/newzvol casesensitivity=sensitive
```
##### Expected Results:

The immutable property should just be skipped, rather than the playbook aborting.
##### Actual Results:

Playbook aborts with error.
##### Other Notes:

As an aside, and happy to make a new issue for this, volblocksize needs to be added to the immutable properties, as this can also only be set at creation time.

Here is a possible patch that I'm currently using (patch against latest git). I'm not sure if it's appropriate though, as I feel I might be missing the reasons for the fail message.

``` patch
diff --git a/system/zfs.py b/system/zfs.py
index 51b9db6..41dcb25 100644
--- a/system/zfs.py
+++ b/system/zfs.py
@@ -244,7 +244,7 @@ class Zfs(object):
         self.properties = properties
         self.changed = False

-        self.immutable_properties = [ 'casesensitivity', 'normalization', 'utf8only' ]
+        self.immutable_properties = [ 'casesensitivity', 'normalization', 'utf8only', 'volblocksize' ]

     def exists(self):
         cmd = [self.module.get_bin_path('zfs', True)]
@@ -324,9 +324,7 @@ class Zfs(object):
         current_properties = self.get_current_properties()
         for prop, value in self.properties.iteritems():
             if current_properties[prop] != value:
-                if prop in self.immutable_properties:
-                    self.module.fail_json(msg='Cannot change property %s after creation.' % prop)
-                else:
+                if not prop in self.immutable_properties:
                     self.set_property(prop, value) 

     def get_current_properties(self):
```

**Comment by robynbergeron** [comment]
[module: system/zfs.py]

**Comment by robynbergeron** [comment]
@techmunk -- Thanks for submitting this bug report.

@johanwiren --  thanks in advance for taking a look at this bug report and for your ongoing work in maintaining this module.

**Comment by johanwiren** [comment]
This should be fixed by #1378

**Comment by ansibot** [comment]
@johanwiren, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by johanwiren** [comment]
resolved_by_pr #1378
